addons/sale_margin/models/sale_order.py

```diff
@@ -60,7 +60,10 @@ def _product_margin(self):
         for line in self:
             currency = line.order_id.pricelist_id.currency_id
             price = line.purchase_price
-            line.margin = currency.round(line.price_subtotal - (price * line.product_uom_qty))
+            if currency:
+                line.margin = currency.round(line.price_subtotal - (price * line.product_uom_qty))
+            else:
+                line.margin = line.price_subtotal - (price * line.product_uom_qty)
```

Comment:
minor: you could do the computation only once, and call the rounding conditionally, to avoid having two copies of the same computation (it looks redundant and could be error-prone if someone forgets to update both later).

i.e.
```
margin = line.price_subtotal - (price * line.product_uom_qty)
line.margin = currency.round(margin) if currency else margin
```

Reply:
:+1: I had the same reaction when reading the code.